api/src/main/java/io/minio/messages/ObjectLockConfiguration.java

```diff
@@ -43,10 +43,10 @@ public ObjectLockConfiguration(RetentionMode mode, RetentionDuration duration)
       throws IllegalArgumentException {
     if (mode != null && duration != null) {
       this.rule = new Rule(mode, duration);
-    }
-
-    if (mode != null || duration != null) {
-      throw new IllegalArgumentException("mode or duration is null");
+    } else if (mode == null) {
+      throw new IllegalArgumentException("mode is null");
+    } else if (duration == null) {
+      throw new IllegalArgumentException("duration is null");
```

Comment:
This is wrong check. You would need to preserve current `if` with `else if` with specific error message is requested
```java
    else if (mode != null || duration != null) {
      if (mode == null) {
        throw new IllegalArgumentException("mode is null");
      }
      throw new IllegalArgumentException("duration is null");
    }
```